eclipse/src/saros/session/resources/validation/ResourceDeltaVisitor.java

```diff
@@ -32,17 +38,22 @@ public boolean visit(final IResourceDelta delta) throws CoreException {
 
     if (resource.getType() == IResource.ROOT) return true;
 
-    if (!session.isShared(ResourceAdapterFactory.create(resource))) return true;
+    saros.filesystem.IResource resourceWrapper =
+        ResourceConverter.convertToResource(referencePoint, resource);
+
+    if (resourceWrapper == null || !session.isShared(resourceWrapper)) return true;
```

Comment:
Isn't there a null check anymore in the isShared method ?

Reply:
Yes, `isShared` still checks for `null`. I am kind of inconsistent on how to handle this. I think for most calls in Saros/I, I still explicitly check for `null` (in case the internal implementation changes?). But I don't think it matter that much. If you want, I can still remove it.